src/test/java/jenkins/plugins/logstash/persistence/ElasticSearchDaoTest.java

```diff
@@ -0,0 +1,127 @@
+    verifyNoMoreInteractions(mockHttpClient);
+    verifyNoMoreInteractions(mockPost);
+    verifyNoMoreInteractions(mockLogger);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void constructorFailNullHost() throws Exception {
+    try {
+      new ElasticSearchDao(null, 8200, "logstash", "username", "password");
+    } catch (IllegalArgumentException e) {
+      assertEquals("Wrong error message was thrown", "host name is required", e.getMessage());
+      throw e;
+    }
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void constructorFailEmptyHost() throws Exception {
+    try {
+      new ElasticSearchDao(" ", 8200, "logstash", "username", "password");
+    } catch (IllegalArgumentException e) {
+      assertEquals("Wrong error message was thrown", "host name is required", e.getMessage());
+      throw e;
+    }
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void constructorFailNullKey() throws Exception {
+    try {
+      new ElasticSearchDao("localhost", 8200, null, "username", "password");
+    } catch (IllegalArgumentException e) {
+      assertEquals("Wrong error message was thrown", "elastic index name is required", e.getMessage());
+      throw e;
+    }
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void constructorFailEmptyKey() throws Exception {
+    try {
+      new ElasticSearchDao("localhost", 8200, " ", "username", "password");
+    } catch (IllegalArgumentException e) {
+      assertEquals("Wrong error message was thrown", "elastic index name is required", e.getMessage());
+      throw e;
+    }
+  }
+
+  @Test
+  public void constructorSuccess1() throws Exception {
+    // Unit under test
+    dao = new ElasticSearchDao("localhost", 8200, "logstash", "username", "password");
+
+    // Verify results
+    assertEquals("Wrong host name", "localhost", dao.host);
+    assertEquals("Wrong port", 8200, dao.port);
+    assertEquals("Wrong key", "logstash", dao.key);
+    assertEquals("Wrong password", "password", dao.password);
+  }
+
+  @Test
+  public void constructorSuccess2() throws Exception {
+    // Unit under test
+    dao = new ElasticSearchDao("localhost", 8200, "jenkins/logstash", "username", "password");
+
+    // Verify results
+    assertEquals("Wrong host name", "localhost", dao.host);
+    assertEquals("Wrong port", 8200, dao.port);
+    assertEquals("Wrong key", "jenkins/logstash", dao.key);
+    assertEquals("Wrong password", "password", dao.password);
+  }
+
+  @Test
+  public void constructorSuccess3() throws Exception {
+    // Unit under test
+    dao = new ElasticSearchDao("localhost", 8200, "/jenkins//logstash/", "username", "password");
+
+    // Verify results
+    assertEquals("Wrong host name", "localhost", dao.host);
+    assertEquals("Wrong port", 8200, dao.port);
+    assertEquals("Wrong key", "/jenkins//logstash/", dao.key);
+    assertEquals("Wrong password", "password", dao.password);
+  }
+}
```

Comment:
Can you add some tests for the `push` method in ElasticSearchDao? (A happy path test at the very minimum.) This will help catch any errors that might be introduced when others modify your code.